src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,150 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
     @Override
     public void add(T value) {
+        add(value, size);
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Wrong index");
+        }
```

Comment:
```suggestion
        if (index < 0 || index > size) {
            throw new IndexOutOfBoundsException("Wrong index " + index " for size " + size);
        }
```

lets make exception messages more informative